client/components/forms/form-radio/index.tsx

```diff
@@ -6,14 +6,21 @@ import './style.scss';
 const FormRadio = ( {
 	className,
 	label,
+	htmlFor,
 	...otherProps
 }: {
 	className?: string;
 	label?: ReactNode;
+	htmlFor?: string;
 } & Omit< HTMLProps< HTMLInputElement >, 'label' > ) => (
 	<>
 		<input { ...otherProps } type="radio" className={ clsx( className, 'form-radio' ) } />
-		{ label && <span className="form-radio__label">{ label }</span> }
+		{ label && ! htmlFor && <span className="form-radio__label">{ label }</span> }
+		{ label && htmlFor && (
+			<label className="form-radio__label" htmlFor={ htmlFor }>
+				{ label }
+			</label>
+		) }
```

Comment:
```suggestion
		{ label && (
			<label className="form-radio__label" { ...( htmlFor && { htmlFor } ) }>
				{ label }
			</label>
		) }
```

Will it be nice to shorten it like this instead of adding the markup twice? WDYT?

Reply:
Hey @Imran92, the problem is that I was not confident removing the span element for the other use cases, as it may cause some styling issues in other parts of Calypso. At first, I tried the approach you suggested, but one of the tests broke.